broker/thinBroker.go

```diff
@@ -1848,7 +1853,7 @@ func (tb *ThinBroker) RegisterCSource(w rest.ResponseWriter, r *rest.Request) {
 				context = append(context, linkMap["rel"]) // Make use of "link" and "type" also
```

Comment:
[Major] No check for id-pattern. As entity registration with id pattern not allowed in fogflow due to limitation of discovery, it should return error

Reply:
added support in code